src/orquestra/sdk/_base/_dsl.py

```diff
@@ -133,7 +133,32 @@ def GithubImport(
     username: Optional[str] = None,
     personal_access_token: Optional[Secret] = None,
 ):
-    """Helper to create GitImports from Github repos"""
+    """
+    Helper to create GitImports from Github repos.
+
+    Raises:
+        TypeError: when a value that is not a `sdk.Secret` is passed as
+            `personal_access_token`.
+    """
+    if personal_access_token is not None and not isinstance(
+        personal_access_token, Secret
+    ):
+        if isinstance(personal_access_token, str):
+            errmsg = (
+                f'You passed a string as `personal_access_token = "..."`. '
+                'Please pass `personal_access_token = sdk.Secret(name="...")` instead. '
+                "It might seem verbose, but it's a precaution against committing "
+                "plain-text credentials to your git repo, or leaking secret values as "
+                "part of the workflow definition."
+                "\nSuggested fix:\n"
+                f'  personal_access_token = sdk.Secret(name="{personal_access_token}")'
```

Comment:
**thought**: I really like providing users with ready to use snippets as suggested fix. However, in this case it might easily lead to leaking credentials in the logs. I'm thinking about a situation where the user mistakenly passes the secret value directly as a string:
```python
_ = sdk.GithubImport(
    ...
    personal_access_token="gh_pat_123_abc123ffff2037deadbeef...",
)
```

```suggestion
                f'  personal_access_token = sdk.Secret(name="paste secret name here")'
```